cmake/modules/FindP4EST.cmake

```diff
@@ -114,7 +114,13 @@ IF(EXISTS ${P4EST_INCLUDE_DIR}/p4est_config.h)
   FILE(STRINGS "${P4EST_INCLUDE_DIR}/p4est_config.h" P4EST_MPI_STRING
     REGEX "#define.*P4EST_MPI 1")
   IF("${P4EST_MPI_STRING}" STREQUAL "")
-    SET(P4EST_WITH_MPI FALSE)
+    FILE(STRINGS "${P4EST_INCLUDE_DIR}/p4est_config.h" P4EST_MPI_STRING
+      REGEX "#define.*P4EST_ENABLE_MPI")
+    IF("${P4EST_MPI_STRING}" STREQUAL "")
+      SET(P4EST_WITH_MPI FALSE)
```

Comment:
Is this even possible? I didn't know you could compile p4est without MPI, and in any case, what would one want to use it for?

Reply:
Yes, it is perfectly possible to compile P4est without MPI support:
  https://github.com/cburstedde/p4est/blob/feature-cmake/cmake/options.cmake#L9
  https://github.com/cburstedde/p4est/blob/master/src/p4est.c#L1297-L1327

P4est without MPI support isn't really useful for us. Because of that we require enabled MPI for P4est.

Reply:
Fascinating. Why would one do that?